p2p/src/protocol.rs

```diff
@@ -568,6 +574,28 @@ impl<T: BlockTree, F: Filters, P: peer::Store> Protocol<T, F, P> {
         }
     }
 
+    /// Send a message to all connected peers. Returns their peer ids.
+    fn broadcast<Q>(&self, msg: NetworkMessage, mut f: Q) -> Option<Vec<PeerId>>
+    where
+        Q: FnMut(&peermgr::Peer) -> bool,
+    {
+        let peers = self
+            .peermgr
+            .outbound()
+            .filter(|p| f(*p))
+            .collect::<Vec<_>>();
+
+        if peers.is_empty() {
+            return None;
+        }
+
+        for peer in &peers {
+            self.upstream.message(peer.address(), msg.clone());
+        }
+
+        Some(peers.iter().map(|p| p.address()).collect())
+    }
```

Comment:
It's cleaner if this function just returns `Vec<_>`, since Vec can be empty. There's no need to wrap it in an `Option`.